samples/browser/src/main/java/org/mozilla/samples/browser/TabsTrayFragment.kt

```diff
@@ -11,35 +11,42 @@ import android.view.ViewGroup
 import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.GridLayoutManager
 import com.google.android.material.snackbar.Snackbar
-import kotlinx.android.synthetic.main.fragment_tabstray.tabsTray
-import kotlinx.android.synthetic.main.fragment_tabstray.toolbar
 import mozilla.components.browser.tabstray.TabsAdapter
 import mozilla.components.browser.thumbnails.loader.ThumbnailLoader
 import mozilla.components.feature.tabs.TabsUseCases
 import mozilla.components.feature.tabs.tabstray.TabsFeature
 import mozilla.components.support.base.feature.UserInteractionHandler
 import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.samples.browser.databinding.FragmentTabstrayBinding
 import org.mozilla.samples.browser.ext.components
 
 /**
  * A fragment for displaying the tabs tray.
  */
-class TabsTrayFragment : Fragment(), UserInteractionHandler {
+class TabsTrayFragment : Fragment(R.layout.fragment_tabstray), UserInteractionHandler {
     private val tabsFeature: ViewBoundFeatureWrapper<TabsFeature> = ViewBoundFeatureWrapper()
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
-        inflater.inflate(R.layout.fragment_tabstray, container, false)
+    private var _binding: FragmentTabstrayBinding? = null
+    // This property is only valid between onCreateView and
+    // onDestroyView.
+    private val binding get() = _binding!!
```

Comment:
I don't understand this pattern. It somehow awkwardly works around the explicitly nullability checks of Kotlin.

Reply:
I think this is a combination of the following two reasons why Google chooses this approach.
1. Avoid leaking views via binding (which I think isn't possible with lateinit) 
2. Using a mutable and nullable property (_binding directly) breaks usability in Kotlin (example https://stackoverflow.com/q/44595529).